include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -73,6 +73,25 @@ def AndOp : UTBinRTLOp<"and", [Commutative]>;
 def OrOp  : UTBinRTLOp<"or",  [Commutative]>;
 def XorOp : UTBinRTLOp<"xor", [Commutative]>;
 
+//===----------------------------------------------------------------------===//
+// Unary Operations
+//===----------------------------------------------------------------------===//
+
+// Base class for unary primitive operators.
+class UnaryRTLOp<string mnemonic, list<OpTrait> traits = []> :
+      RTLOp<mnemonic, !listconcat(traits, [NoSideEffect])> {
+  let arguments = (ins AnySignlessInteger:$input);
+  let results = (outs AnySignlessInteger:$result);
+
+  let assemblyFormat = [{
+    $input attr-dict `:` type($input) `,` type($result)
```

Comment:
Should this be functional-type instead of relying on the attribute dictionar?

Reply:
fixed with https://github.com/circt/circt/pull/11/commits/ecf7fdf751d0e2c407ef71efd42725874ba6550a